client/src/components/HomeScreen/ModalShareExpenses.tsx

```diff
@@ -0,0 +1,244 @@
+import { Dispatch, SetStateAction, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import {
+  Modal,
+  Pressable,
+  StyleSheet,
+  Text,
+  TextInput,
+  View,
+} from 'react-native';
+import * as Yup from 'yup';
+import { Formik, FormikValues } from 'formik';
+import { LOCALES_EN } from '../../constants/locales/en';
+import { STYLES } from '../../styles/styles';
+import { COLORS } from '../../styles/colors';
+import { shareExpense } from '../../helpers/api';
+import { useStore } from '../../mobx/store';
+
+type ModalShareExpensesProps = {
+  modalShareExpensesVisible: boolean;
+  setModalShareExpensesVisible: Dispatch<SetStateAction<boolean>>;
+};
+
+const initialValues = {
+  email: '',
+  sum: '0',
+};
+
+const ValidationSchema = Yup.object().shape({
+  email: Yup.string().email('Invalid email').required('Required'),
+  sum: Yup.string()
+    .matches(/^\d{1,15}(\.\d+)?$/, 'Enter a valid number')
+    .required('Required'),
+});
+
+export default function ModalShareExpenses({
+  modalShareExpensesVisible,
+  setModalShareExpensesVisible,
+}: ModalShareExpensesProps) {
+  const [error, setError] = useState('');
+  const [success, setSuccess] = useState('');
+  const { currentUserId } = useStore();
+
+  const handleFormSubmit = async (values: FormikValues) => {
+    const numberSum = +parseFloat(values.sum).toFixed(3);
+    try {
+      const result = await shareExpense(currentUserId, values.email, numberSum);
+      setSuccess(result);
+      //setModalShareExpensesVisible(!modalShareExpensesVisible);
+    } catch (err: unknown) {
+      if (err instanceof Error) {
+        setError(err.message);
+      }
+    }
+  };
+  return (
+    <Modal
+      animationType="fade"
+      transparent={true}
+      visible={modalShareExpensesVisible}
+      onRequestClose={() => {
+        setModalShareExpensesVisible(!modalShareExpensesVisible);
+      }}
+    >
+      <View style={styles.centeredView}>
```

Comment:
It's a good pattern to reuse styles, but it seems overkill to use multiple views to adjust position, you could also combine styles for one view, I believe in this case it will have some adjustments